torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -264,6 +264,36 @@ std::shared_ptr<SugaredValue> ClassValue::attr(
   }
   return std::make_shared<ClassNewMethod>(type_);
 }
+
+std::shared_ptr<SugaredValue> NamedTupleValue::call(
+    const SourceRange& loc,
+    Function& m,
+    at::ArrayRef<NamedValue> inputs,
+    at::ArrayRef<NamedValue> attributes,
+    size_t n_binders) {
+  auto nargs = type_->containedTypes().size();
+  if (inputs.size() != nargs) {
+    throw ErrorReport(loc) << "Constructor expected " << nargs << "arguments "
+                           << "but got " << inputs.size();
+  }
+  AT_CHECK(attributes.size() == 0);
+
+  auto& g = *m.graph();
+
+  std::vector<Value*> input_vals;
+  for (auto& inp : inputs) {
+    input_vals.push_back(inp.value(g));
+  }
+
+  auto self = g.insertNode(g.createTuple(
+                                input_vals, type_->names(), type_->unqualName())
+                               ->setSourceRange(loc))
+                  ->output();
+  self->setType(type_);
```

Comment:
Uhhh, there is absolutely no type checking here!

Reply:
I suggest that a FunctionSchema is created that is the right constructor list for the named tuple, and then matchSchema is used. That fixes the keyword arguments, and does the type checking correctly.